src/anti_patterns/borrow_clone.md

```diff
@@ -36,13 +36,10 @@ between the two are not synchronized -- as if two completely separate variables
 exist.
 
 There are special cases -- `Rc<T>` is designed to handle clones intelligently.
-It internally manages exactly one copy of the data, and cloning it will only
-clone the reference.
-
-There is also `Arc<T>` which provides shared ownership of a value of type T that
-is allocated in the heap. Invoking `.clone()` on `Arc` produces a new `Arc`
-instance, which points to the same allocation on the heap as the source `Arc`,
-while increasing a reference count.
+It internally manages exactly one copy of the data. Invoking `.clone()` on `Rc`
+produces a new `Rc` instance, which points to the same data as the source `Rc`,
+while increasing a reference count. The same applies to `Arc`, the thread-safe
+counterpart of `Rc`.
```

Comment:
It sounded like the use of heap is specific to `Arc`. I believe the use of heap is irrelevant here. I combined the best parts of `Rc` and `Arc` paragraphs and mentioned `Arc` at the end.